types/index.d.ts

```diff
@@ -0,0 +1,49 @@
+import * as V from './vue';
+import * as Options from './options';
+import * as Plugin from './plugin';
+
+declare global {
+  interface Array<T> {
+    $remove(item: T): Array<T>;
+    $set(index: any, val: T): T;
+  }
+
+  // For the projects/tools that depend on this namespace
+  namespace vuejs {
+    export type PropOption = Options.PropOptions;
+    export type ComputedOption = Options.ComputedOptions<any>;
```

Comment:
Considering the backward compatibility, I couldn't do so...
It's because the current definition for Vue 1.x didn't define `this` as `V.Vue`.
How do you think?